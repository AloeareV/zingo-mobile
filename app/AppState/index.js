import Address from './Address';
import AddressBookEntry from './AddressBookEntry';
import AppStateLoaded from './AppStateLoaded';
import AppStateLoading from './AppStateLoading';
import ConnectedCompanionApp from './ConnectedCompanionApp';
import ErrorModalData from './ErrorModalData';
import InfoType from './InfoType';
import PasswordState from './PasswordState';
import ReceivePageState from './ReceivePageState';
import SendJsonToType from './SendJsonToType';
import SendPageState from './SendPageState';
import SendProgress from './SendProgress';
import ServerSelectState from './ServerSelectState';
import SettingsFileEntry from './SettingsFileEntry';
import SyncStatus from './SyncStatus';
import SyncStatusReport from './SyncStatusReport';
import ToAddr from './ToAddr';
import TotalBalance from './TotalBalance';
import Transaction from './Transaction';
import TxDetailType from './TxDetailType';
import WalletSeed from './WalletSeed';
import WalletSettings from './WalletSettings';

export {
  AppStateLoaded,
  AppStateLoading,
  Address,
  AddressBookEntry,
  ConnectedCompanionApp,
  ErrorModalData,
  InfoType,
  PasswordState,
  ReceivePageState,
  SendJsonToType,
  SendPageState,
  SendProgress,
  ServerSelectState,
  SettingsFileEntry,
  SyncStatus,
  SyncStatusReport,
  ToAddr,
  TotalBalance,
  Transaction,
  TxDetailType,
  WalletSeed,
  WalletSettings,
};
