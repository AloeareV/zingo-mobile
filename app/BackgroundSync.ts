import RPCModule from './RPCModule';
import AsyncStorage from '@react-native-async-storage/async-storage';
import NetInfo, { NetInfoStateType } from '@react-native-community/netinfo';
import { RPCSyncStatusType } from './rpc/types/RPCSyncStatusType';

const BackgroundSync = async (task_data: any) => {
  // this not interact with the server.
  const exists = await RPCModule.walletExists();

  // only if exists the wallet file make sense to do the sync.
  if (exists && exists !== 'false') {
    // only if we have connection make sense to call RPCModule.
    const networkState = await NetInfo.fetch();
    if (
      !networkState.isConnected ||
      networkState.type === NetInfoStateType.cellular ||
      (networkState.details !== null && networkState.details.isConnectionExpensive)
    ) {
      //console.log(
      //  'BS: Not started (connected: ' + networkState.isConnected,
      //  +', type: ' +
      //    networkState.type +
      //    +', expensive connection: ' +
      //    networkState.details?.isConnectionExpensive +
      //    ')',
      //);
      return;
    }
    // if the App goes to Foreground kill the interval
    const background = await AsyncStorage.getItem('@background');
    if (background === 'no') {
      //console.log('BS: Not started (going to foreground)');
      return;
    }

    let batch_num = -1;
    console.log('BS:', task_data);

    // finishEarly has two fields: wait, and done.
    // wait() returns a promise, which is resolved when
    // done() is called
    let finishEarly = manuallyResolve();

    let saver = setInterval(async () => {
      const networkStateSaver = await NetInfo.fetch();
      if (
        !networkStateSaver.isConnected ||
        networkStateSaver.type === NetInfoStateType.cellular ||
        (networkStateSaver.details !== null && networkStateSaver.details.isConnectionExpensive)
      ) {
        //console.log(
        //  'BS: Interrupted (connected: ' + networkStateSaver.isConnected,
        //  +', type: ' +
        //    networkStateSaver.type +
        //    +', expensive connection: ' +
        //    networkStateSaver.details?.isConnectionExpensive +
        //    ')',
        //);
        clearInterval(saver);
        finishEarly.done();
        return;
      }
      // if the App goes to Foreground kill the interval
      const backgroundSaver = await AsyncStorage.getItem('@background');
      if (backgroundSaver === 'no') {
        clearInterval(saver);
        //console.log('BS: Finished (going to foreground)');
        finishEarly.done();
        return;
      }

      const syncStatusStr: string = await RPCModule.execute('syncstatus', '');
      if (syncStatusStr) {
        if (syncStatusStr.toLowerCase().startsWith('error')) {
          //console.log(`BS: Error sync status ${syncStatusStr}`);
          return;
        }
      } else {
        //console.log('BS: Internal Error sync status');
        return;
      }
      // TODO: verify this JSON parse
      const ss: RPCSyncStatusType = await JSON.parse(syncStatusStr);

      //console.log('BS:', ss);
      if (ss.batch_num && ss.batch_num > -1 && batch_num !== ss.batch_num) {
        await RPCModule.doSave();
        //console.log('BS: saving...');
        // update batch_num with the new value, otherwise never change
        batch_num = ss.batch_num;
      }
    }, 5000);

    await Promise.race([RPCModule.execute('sync', ''), finishEarly.wait()]);
    clearInterval(saver);
  } else {
    console.log('BS: wallet file does not exist');
  }
  //console.log('BS: Finished (end of syncing)');
};

export default BackgroundSync;

function manuallyResolve() {
  let resolve: Function;
  // new Promise takes a function as an argument. When that function is called
  // the promise resolves with the value output by that function.
  // By passing the function out of the promise, we can call it later
  // in order to resolve the promise at will
  const promise = new Promise(fun => {
    resolve = fun;
  });

  function done() {
    resolve();
  }

  function wait() {
    return promise;
  }

  return { wait, done };
}
