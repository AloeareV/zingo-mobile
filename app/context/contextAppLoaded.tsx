import React, { ReactNode } from 'react';
import { StackScreenProps } from '@react-navigation/stack';

import {
  AppStateLoaded,
  SyncingStatusReportClass,
  TotalBalanceClass,
  ReceivePageStateClass,
  ErrorModalDataClass,
  SendProgressClass,
  WalletSettingsClass,
  TransactionType,
  InfoType,
  SyncingStatusType,
  WalletSeedType,
  DimensionsType,
  AddressClass,
  AddressBookClass,
  zecPriceType,
  BackgroundType,
  SendPageStateClass,
  ToAddrClass,
  NetInfoType,
  BackgroundErrorType,
} from '../AppState';

export const defaultAppStateLoaded: AppStateLoaded = {
  navigation: {} as StackScreenProps<any>['navigation'],
  route: {} as StackScreenProps<any>['route'],
  dimensions: {} as DimensionsType,
  appState: '' as string,
  netInfo: {} as NetInfoType,

  syncingStatusReport: new SyncingStatusReportClass(),
  totalBalance: new TotalBalanceClass(),
  addressPrivateKeys: new Map(),
  addresses: [] as AddressClass[],
  addressBook: [] as AddressBookClass[],
  transactions: [] as TransactionType[],
  sendPageState: new SendPageStateClass(new ToAddrClass(0)),
  receivePageState: new ReceivePageStateClass(''),
  info: {} as InfoType,
  walletSettings: new WalletSettingsClass(),
  syncingStatus: {} as SyncingStatusType,
  errorModalData: new ErrorModalDataClass('', ''),
  sendProgress: new SendProgressClass(0, 0, 0),
  walletSeed: {} as WalletSeedType,
  isMenuDrawerOpen: false,
  selectedMenuDrawerItem: '' as string,
  aboutModalVisible: false,
  computingModalVisible: false,
  settingsModalVisible: false,
  infoModalVisible: false,
  rescanModalVisible: false,
  seedViewModalVisible: false,
  seedChangeModalVisible: false,
  seedBackupModalVisible: false,
  seedServerModalVisible: false,
  syncReportModalVisible: false,
  poolsModalVisible: false,
  newServer: '' as string,
  uaAddress: '' as string,

  server: '' as string,
  currency: '',
  language: 'en',

  zecPrice: {
    zecPrice: 0,
    date: 0,
  } as zecPriceType,
  sendAll: false,
  background: {
    batches: 0,
    date: 0,
  } as BackgroundType,

  translate: () => '',
  backgroundError: {} as BackgroundErrorType,
};

export const ContextAppLoaded = React.createContext(defaultAppStateLoaded);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoaded;
};

export const ContextAppLoadedProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextAppLoaded.Provider value={value}>{children}</ContextAppLoaded.Provider>;
};
