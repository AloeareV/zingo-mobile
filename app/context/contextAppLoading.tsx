import React, { ReactNode } from 'react';
import { StackScreenProps } from '@react-navigation/stack';

import {
  AppStateLoading,
  DimensionsType,
  InfoType,
  TotalBalanceClass,
  WalletSeedType,
  zecPriceType,
  backgroundType,
} from '../AppState';

export const defaultAppStateLoading: AppStateLoading = {
  navigation: {} as StackScreenProps<any>['navigation'],
  route: {} as StackScreenProps<any>['route'],
  dimensions: {} as DimensionsType,
  appState: '' as string,

  screen: 0,
  actionButtonsDisabled: false,
  walletExists: false,
  walletSeed: {} as WalletSeedType,
  totalBalance: new TotalBalanceClass(),
  info: {} as InfoType,

  server: '' as string,
  currency: '',
  language: 'en',

  zecPrice: {
    zecPrice: 0,
    date: 0,
  } as zecPriceType,
  sendAll: false,
  background: {
    batches: 0,
    date: 0,
  } as backgroundType,

  translate: () => '',
};

export const ContextAppLoading = React.createContext(defaultAppStateLoading);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoading;
};

export const ContextAppLoadingProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextAppLoading.Provider value={value}>{children}</ContextAppLoading.Provider>;
};
