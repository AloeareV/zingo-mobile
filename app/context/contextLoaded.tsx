import React, { ReactNode } from 'react';

import {
  AppStateLoaded,
  SyncStatusReport,
  TotalBalance,
  SendPageState,
  ReceivePageState,
  ToAddr,
  ErrorModalData,
  SendProgress,
  WalletSettings,
} from '../AppState';

const defaultAppState: AppStateLoaded = {
  navigation: null,
  route: null,
  dimensions: {} as {
    width: number;
    height: number;
    orientation: 'portrait' | 'landscape';
    deviceType: 'tablet' | 'phone';
    scale: number;
  },

  syncStatusReport: new SyncStatusReport(),
  totalBalance: new TotalBalance(),
  addressPrivateKeys: new Map(),
  addresses: [],
  addressBook: [],
  transactions: null,
  sendPageState: new SendPageState(new ToAddr(0)),
  receivePageState: new ReceivePageState(),
  info: null,
  rescanning: false,
  wallet_settings: new WalletSettings(),
  syncingStatus: null,
  errorModalData: new ErrorModalData(),
  txBuildProgress: new SendProgress(),
  walletSeed: null,
  isMenuDrawerOpen: false,
  selectedMenuDrawerItem: '',
  aboutModalVisible: false,
  computingModalVisible: false,
  settingsModalVisible: false,
  infoModalVisible: false,
  rescanModalVisible: false,
  seedViewModalVisible: false,
  seedChangeModalVisible: false,
  seedBackupModalVisible: false,
  seedServerModalVisible: false,
  syncReportModalVisible: false,
  poolsModalVisible: false,
  newServer: null,
  uaAddress: null,

  translate: () => {},
  openErrorModal: () => {},
  closeErrorModal: () => {},
  toggleMenuDrawer: () => {},
  setComputingModalVisible: () => {},
  syncingStatusMoreInfoOnClick: async () => {},
  poolsMoreInfoOnClick: async () => {},
  startRescan: () => {},
};

export const ContextLoaded = React.createContext(defaultAppState);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoaded;
};

export const ContextLoadedProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextLoaded.Provider value={value}>{children}</ContextLoaded.Provider>;
};
