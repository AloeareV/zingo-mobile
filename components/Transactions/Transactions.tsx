import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';

import TransactionsView from './components/TransactionsView';

const Stack = createStackNavigator();

type TransactionsViewProps = {
  info: Info | null;
  totalBalance: TotalBalance;
  syncingStatus: SyncStatus | null;
  transactions: Transaction[] | null;
  toggleMenuDrawer: () => void;
  doRefresh: () => void;
  setComputingModalVisible: (visible: boolean) => void;
};

const Transactions: React.FunctionComponent<TransactionsViewProps> = iprops => {
  return (
    <Stack.Navigator screenOptions={{ headerShown: false }}>
      <Stack.Screen name="TransactionsView">{props => <TransactionsView {...props} {...iprops} />}</Stack.Screen>
    </Stack.Navigator>
  );
};

export default Transactions;
