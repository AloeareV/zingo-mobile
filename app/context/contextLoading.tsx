import React, { ReactNode } from 'react';
import { StackScreenProps } from '@react-navigation/stack';

import { AppStateLoading, DimensionsType, InfoType, TotalBalanceClass, WalletSeedType } from '../AppState';

export const defaultAppStateLoading: AppStateLoading = {
  navigation: {} as StackScreenProps<any>['navigation'],
  route: {} as StackScreenProps<any>['route'],
  dimensions: {} as DimensionsType,

  screen: 0,
  actionButtonsDisabled: false,
  walletExists: false,
  walletSeed: {} as WalletSeedType,
  server: '' as string,
  currency: '',
  language: 'en',
  totalBalance: new TotalBalanceClass(),
  info: {} as InfoType,

  translate: () => '',
};

export const ContextLoading = React.createContext(defaultAppStateLoading);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoading;
};

export const ContextLoadingProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextLoading.Provider value={value}>{children}</ContextLoading.Provider>;
};
