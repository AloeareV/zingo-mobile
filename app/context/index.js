import { defaultAppStateLoaded, ContextAppLoaded, ContextAppLoadedProvider } from './contextAppLoaded';
import { defaultAppStateLoading, ContextAppLoading, ContextAppLoadingProvider } from './contextAppLoading';

export {
  defaultAppStateLoaded,
  ContextAppLoaded,
  ContextAppLoadedProvider,
  defaultAppStateLoading,
  ContextAppLoading,
  ContextAppLoadingProvider,
};
