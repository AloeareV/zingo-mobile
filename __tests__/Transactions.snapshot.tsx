/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import Transactions from '../components/Transactions';
import { ContextLoadedProvider } from '../app/context';

import {
  ErrorModalData,
  ReceivePageState,
  SendPageState,
  SendProgress,
  SyncStatusReport,
  ToAddr,
  TotalBalance,
  WalletSettings,
} from '../app/AppState';

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.', // us
      groupingSeparator: ',', // us
    };
  },
}));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');


// test suite
describe('Component Receive - test', () => {
  //snapshot test
  test('Receive - snapshot', () => {
    const state = {
      navigation: null,
      route: null,

      syncStatusReport: new SyncStatusReport(),
      addressPrivateKeys: new Map(),
      addressBook: [],
      transactions: null,
      sendPageState: new SendPageState(new ToAddr(0)),
      receivePageState: new ReceivePageState(),
      info: null,
      rescanning: false,
      wallet_settings: new WalletSettings(),
      syncingStatus: null,
      errorModalData: new ErrorModalData(),
      txBuildProgress: new SendProgress(),
      walletSeed: null,
      isMenuDrawerOpen: false,
      selectedMenuDrawerItem: '',
      aboutModalVisible: false,
      computingModalVisible: false,
      settingsModalVisible: false,
      infoModalVisible: false,
      rescanModalVisible: false,
      seedViewModalVisible: false,
      seedChangeModalVisible: false,
      seedBackupModalVisible: false,
      seedServerModalVisible: false,
      syncReportModalVisible: false,
      poolsModalVisible: false,
      newServer: null,
      uaAddress: 'UA',
      addresses: [
        {
          uaAddress: 'UA',
          address: 'UA',
          addressKind: 'u',
          containsPending: false,
          receivers: 'ozt',
        },
      ],
      translate: () => 'text translated',
      dimensions: {
        width: 200,
        height: 200,
        orientation: 'portrait',
        deviceType: 'tablet',
        scale: 1.5,
      } as {
        width: number;
        height: number;
        orientation: 'portrait' | 'landscape';
        deviceType: 'tablet' | 'phone';
        scale: number;
      },
      totalBalance: new TotalBalance(),
    };
    const receive = render(
      <ContextLoadedProvider value={state}>
        <Transactions
          doRefresh={() => {}}
          toggleMenuDrawer={() => {}}
          setComputingModalVisible={() => {}}
          poolsMoreInfoOnClick={() => {}}
          syncingStatusMoreInfoOnClick={() => {}}
        />
      </ContextLoadedProvider>,
    );
    expect(receive.toJSON()).toMatchSnapshot();
  });
});
