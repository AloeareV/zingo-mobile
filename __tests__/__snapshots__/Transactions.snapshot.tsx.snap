// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Receive - test Receive - snapshot 1`] = `
<View
  accessibilityLabel="text translated"
  accessible={true}
  style={
    {
      "display": "flex",
      "justifyContent": "flex-start",
      "marginBottom": 140,
      "width": "100%",
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(255, 255, 255)",
        "display": "flex",
        "paddingBottom": 0,
        "paddingTop": 10,
        "zIndex": -1,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/img/logobig-zingo.png",
        }
      }
      style={
        {
          "height": 80,
          "resizeMode": "contain",
          "width": 80,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "baseline",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 36,
            }
          }
        >
          á™‡
        </Text>
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 36,
              "fontWeight": "700",
            }
          }
        >
           0.0000
        </Text>
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 18,
              "paddingBottom": 0,
            }
          }
        >
          0000
        </Text>
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 36,
            }
          }
        >
           ---
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "baseline",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "color": undefined,
            "fontSize": 20,
            "marginBottom": 5,
            "marginTop": 0,
          }
        }
      >
        $
      </Text>
      <Text
        style={
          {
            "color": undefined,
            "fontSize": 20,
            "fontWeight": "700",
            "marginBottom": 5,
            "marginTop": 0,
          }
        }
      >
         --
      </Text>
      <Text
        style={
          {
            "color": undefined,
            "fontSize": 20,
            "marginBottom": 5,
            "marginTop": 0,
          }
        }
      >
         USD
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginVertical": 5,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "flexWrap": "wrap",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            [
              {
                "paddingHorizontal": 5,
              },
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "fontWeight": "600",
                "opacity": 1,
              },
            ]
          }
        >
          text translated
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "rgb(255, 255, 255)",
        "padding": 10,
        "position": "absolute",
      }
    }
  >
    <View
      accessibilityLabel="text translated"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <
        color="rgb(216, 216, 216)"
        icon={
          {
            "icon": [
              448,
              512,
              [
                "navicon",
              ],
              "f0c9",
              "M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z",
            ],
            "iconName": "bars",
            "prefix": "fas",
          }
        }
        size={48}
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "flex-end",
        "padding": 10,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Text
      style={
        {
          "color": "rgb(216, 216, 216)",
          "fontSize": 8,
        }
      }
    >
      (200x200)-1.5
    </Text>
    <Text
      style={
        {
          "color": "rgb(216, 216, 216)",
          "fontSize": 7,
        }
      }
    >
      text translated
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "rgb(0, 122, 255)",
        "height": 1,
        "width": "100%",
      }
    }
  />
  <RCTScrollView
    accessibilityLabel="text translated"
    accessible={true}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
        tintColor="rgb(28, 28, 30)"
        title="text translated"
      />
    }
    style={
      {
        "flexGrow": 1,
        "height": "100%",
        "marginTop": 10,
        "width": "100%",
      }
    }
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;
